vector_squared_k_support.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+
+
+
+def prox(v, beta, k):
```

Comment:
This file also has a bunch of functions with the same names as in `l1.py`, which means it's probably another good candidate for being wrapped into a class.